tests/codegen/strncmp.cpp

```diff
@@ -0,0 +1,92 @@
+#include "common.h"
+
+namespace bpftrace {
+namespace test {
+namespace codegen {
+
+TEST(codegen, strncmp_test)
+{
+  test("kretprobe:vfs_read /strncmp(comm, \"sshd\", 2)/ { @[comm] = count(); }",
+
+R"EXPECTED(; Function Attrs: nounwind
```

Comment:
Do I need to have a LLVM < 6 block for this tests (the one I was emulating had it). I don't have a very easy access to older llvms, but I can try and get it if needed.

Reply:
I can run the tests on the LLVM versions failing on CI (I have them easily set up on my machine). Can you rebase and fix the `if` expression issue first? The code generated might change a little bit after that.

Reply:
@mmarchini  do you have a moment to look at the LLVM5 test here?

Reply:
This seems to be the same issue as #186. 

It is fine with me if you add this to the excludes for the LLVM5 build